src/RemoteInboxNotifications/SpecRunner.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Runs a single spec.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+/**
+ * Runs a single spec.
+ */
+class SpecRunner {
+	/**
+	 * Run the spec.
+	 *
+	 * @param object $spec The spec to run.
+	 */
+	public static function run_spec( $spec ) {
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		// Create or update the note.
+		$existing_note_ids = $data_store->get_notes_with_name( $spec->slug );
+		if ( 0 === count( $existing_note_ids ) ) {
+			$note = new WC_Admin_Note();
```

Comment:
It may be worthwhile to add in a new note class `WC_Admin_Rind_Note()` that makes use of note traits introduced in https://github.com/woocommerce/woocommerce-admin/pull/4340.

This would mean not needing to check against existing notes, notes of marketing type, and the evaluation rules could be added as a condition to the `possibly_add_note()` method.

If you have any other ideas or challenges with how you think this should be implemented, let me know and I can help revise note traits to better fit the notes here.